airbyte-integrations/connectors/destination-xata/destination_xata/destination.py

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+from typing import Any, Iterable, Mapping
+
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.destinations import Destination
+from airbyte_cdk.models import AirbyteConnectionStatus, AirbyteMessage, ConfiguredAirbyteCatalog, Status, Type
+
+from xata.client import XataClient
+from xata.helpers import BulkProcessor
+
+__version__ = "0.0.1"
+
+class DestinationXata(Destination):
+    def write(
+        self, config: Mapping[str, Any], configured_catalog: ConfiguredAirbyteCatalog, input_messages: Iterable[AirbyteMessage]
+    ) -> Iterable[AirbyteMessage]:
+        """
+        Reads the input stream of messages, config, and catalog to write data to the destination.
+
+        This method returns an iterable (typically a generator of AirbyteMessages via yield) containing state messages received
+        in the input message stream. Outputting a state message means that every AirbyteRecordMessage which came before it has been
+        successfully persisted to the destination. This is used to ensure fault tolerance in the case that a sync fails before fully completing,
+        then the source is given the last state message output from this method as the starting point of the next sync.
+
+        :param config: dict of JSON configuration matching the configuration declared in spec.json
+        :param configured_catalog: The Configured Catalog describing the schema of the data being received and how it should be persisted in the
+                                    destination
+        :param input_messages: The stream of input messages received from the source
+        :return: Iterable of AirbyteStateMessages wrapped in AirbyteMessage structs
+        """
+
+        xata = XataClient(api_key=config["api_key"], db_url=config["db_url"])
+        xata.set_header("user-agent", f"airbyte/destination-xata:{__version__}")
+        
+        bp = BulkProcessor(xata)
+        count = 0
+        for message in input_messages:
+            if message.type == Type.RECORD:
+                # Put record to processing queue
+                bp.put_record(message.record.stream, message.record.data)
+                count += 1
+            if message.type == Type.STATE:
+                yield message
+        bp.flush_queue()
+        print(f"total: {count}")
+        print(bp.get_stats())
+        if count != bp.get_stats()["total"] or bp.get_stats()["failed_batches"] != 0:
```

Comment:
Please move the print stmt to self.logger.info

Reply:
done via https://github.com/airbytehq/airbyte/pull/24192/commits/ce028d18cc155be6b7e00ecfca69ba341a4cb9fb